src/include/btmem.h

```diff
@@ -410,10 +409,24 @@ struct __wt_page {
 				uint32_t entries;
 				WT_REF	**index;
 			} * volatile __index;	/* Collated children */
+
+			/*
+			 * When splitting to deepen the tree, track the number
+			 * of entries in the newly created parent, and how many
+			 * subsequent splits follow the initial set of entries.
+			 * If future splits into the page are generally after
+			 * the initial set of items, perform future deepening
+			 * splits in this page to optimize for an append-style
+			 * workload.
+			 */
+			uint32_t deepen_split_append;
+			uint32_t deepen_split_last;
```

Comment:
FWIW, I think these could go into `WT_PAGE_MODIFY->u2.intl` in order to keep clean pages as small as possible.

An alternative approach would be to keep a running average "split point percentage" -- each time we split into an internal page, calculate `split_pct = 100 * split_point / entries`.  Then something `avg_split_pct = 0.9 * avg_split_pct + 0.1 * split_pct`.

For append-only workloads, this would be close to 100, but it might be easier to generalize.